src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -590,19 +590,16 @@ internal int BytesToWrite
         internal SerialStream(string portName, int baudRate, Parity parity, int dataBits, StopBits stopBits, int readTimeout, int writeTimeout, Handshake handshake,
             bool dtrEnable, bool rtsEnable, bool discardNull, byte parityReplace)
         {
-            int flags = NativeMethods.FILE_FLAG_OVERLAPPED;
-
-            if ((portName == null) || !portName.StartsWith("COM", StringComparison.OrdinalIgnoreCase))
+            if ((portName == null) ||
+                !portName.StartsWith("COM", StringComparison.OrdinalIgnoreCase) || 
+                !uint.TryParse(portName.Substring(3), out uint portNumber))
+            {
                 throw new ArgumentException(SR.Arg_InvalidSerialPort, nameof(portName));
+            }
 
             // Error checking done in SerialPort.
 
-            SafeFileHandle tempHandle = Interop.Kernel32.CreateFileDefaultSecurity(
-                @"\\?\" + portName,
-                Interop.Kernel32.GenericOperations.GENERIC_READ | Interop.Kernel32.GenericOperations.GENERIC_WRITE,
-                0,              // comm devices must be opened w/exclusive-access
-                FileMode.Open,  // comm devices must use OPEN_EXISTING
-                flags);
+            SafeFileHandle tempHandle = SerialCommunication.Current.OpenPort(portNumber);
```

Comment:
It looks like we build two flavors, one for UWP and one for Win32, and each gets its own SerialCommunication.Current instance, which just supplies a virtual OpenPort method.  Instead, would it make sense to just make SerialStream a partial class, and have a partial class for each of UWP and Win32 that supplies the non-virtual OpenPort method?  That's what we've generally done elsewhere in such situations.